game-app/game-core/src/main/java/games/strategy/triplea/ui/TooltipProperties.java

```diff
@@ -2,52 +2,23 @@
 
 import games.strategy.engine.data.GamePlayer;
 import games.strategy.engine.data.UnitType;
-import games.strategy.triplea.ResourceLoader;
 import games.strategy.triplea.attachments.UnitAttachment;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URL;
-import java.time.Instant;
-import java.util.Optional;
-import java.util.Properties;
 import lombok.extern.slf4j.Slf4j;
-import org.triplea.java.UrlStreams;
 import org.triplea.util.LocalizeHtml;
 
 /** Generates unit tooltips based on the content of the map's {@code tooltips.properties} file. */
 @Slf4j
-public final class TooltipProperties {
-  // Filename
+public final class TooltipProperties extends PropertyFile {
   private static final String PROPERTY_FILE = "tooltips.properties";
-  // Properties
   private static final String TOOLTIP = "tooltip";
   private static final String UNIT = "unit";
-  private static TooltipProperties ttp = null;
-  private static Instant timestamp = Instant.EPOCH;
-  private final Properties properties = new Properties();
 
   private TooltipProperties() {
-    final ResourceLoader loader = UiContext.getResourceLoader();
-    final URL url = loader.getResource(PROPERTY_FILE);
-    if (url != null) {
-      final Optional<InputStream> inputStream = UrlStreams.openStream(url);
-      if (inputStream.isPresent()) {
-        try {
-          properties.load(inputStream.get());
-        } catch (final IOException e) {
-          log.error("Error reading: " + PROPERTY_FILE, e);
-        }
-      }
-    }
+    super(PROPERTY_FILE);
   }
 
   public static TooltipProperties getInstance() {
-    // cache properties for 5 seconds
-    if (ttp == null || timestamp.plusSeconds(5).isBefore(Instant.now())) {
```

Comment:
There's a small change in behaviour here, the cache timeout is now 10 seconds, but having a look at the code any timeout seems ridiculous, it's probably to avoid over-using the IO but it's very brittle to be honest

Reply:
We may want/need this cache. You potentially will observe a slow-down on systems with a slower hard disk and during AI play. TheDog was timing AI play on age of tribes and was seeing a noticable slow down that was in part related to this method.

Why it is possibly needed - tooltip is eagerly created for all swing components even if not rendered. That means on any screen draw we're going to be reading hard disk quite a lot for a lot of the on-map entities. Maps like 'age of tribes' that have lots of units and lots of tooltip possibilities I think are where you will see the problem the most.

We'll need to consider this, I generally agree the cache and the way it is done is not ideal. The only way I know to really check this out is to measure AI play on age of tribes over a round or two. You might be able to get some help getting those measurements in the related issue thread: https://github.com/triplea-game/triplea/issues/9001

Reply:
Thinking about this one, we can use PerfTimer to verify the cost of this method and how often it's executed and exactly how much the cache is saving us. We need to keep in mind slower hard disks could be quite a bit slower executing the IO.

Reply:
I think intelliJs profiling tools could help here too, I'll try to get some numbers here

Reply:
Hmm I have the suspicion that my system isn't really a good machine to test this on.
I checked how long the constructors took to execute and it took less than 2ms every time, probably because I'm using an SSD. So unless this is called in a loop I doubt that I will measure a significant difference on my machine :/

Reply:
Any luck profiling how long this method takes? Even if your system is fast, I think results that indicate if this a NS or a MS operation are going to be useful. If NS then we are likely fine even if it's 3 magnitudes slower.